log-manager/src/test/java/io/airlift/log/TestSocketHandler.java

```diff
@@ -0,0 +1,51 @@
+package io.airlift.log;
+
+import com.google.common.io.ByteStreams;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+import static io.airlift.log.Format.TEXT;
+import static io.airlift.log.SocketMessageOutput.createSocketHandler;
+import static org.testng.Assert.assertEquals;
+
+@Test(timeOut = 5 * 60 * 1000)
+public class TestSocketHandler
+{
+    @Test
+    public void testSocketLogging()
```

Comment:
Try to add a test that connects to an unallocated socket and verify you get the error handling you want. (I'm not sure you can do this)

Reply:
It works, but the errors suck, I'll fix this tomorrow morning.